src/main/kotlin/org/dif/message/Message.kt

```diff
@@ -0,0 +1,110 @@
+package org.dif.message
+
+import org.dif.common.JSON
+import org.dif.common.Typ
+
+data class Message(
+    val id: String,
+    val payload: JSON,
+    val type: String,
+    val typ: Typ,
+    val from: String?,
+    val to: List<String>?,
+    val createdTime: Int?,
+    val expiresTime: Int?,
+    val headers: Map<String, Any>?,
+    val fromPrior: FromPrior?,
+    val attachments: List<Attachment>?,
+    val pleaseAck: Boolean?,
+    val ack: String?,
+    val thid: String?,
+    val pthid: String?
+) {
+    private constructor(builder: Builder) : this(
+        builder.id,
+        builder.payload,
+        builder.type,
+        builder.typ,
+        builder.from,
+        builder.to,
+        builder.createdTime,
+        builder.expiresTime,
+        builder.headers,
+        builder.fromPrior,
+        builder.attachments,
+        builder.pleaseAck,
+        builder.ack,
+        builder.thid,
+        builder.pthid
+    )
+
+    companion object {
+        fun builder(id: String, payload: JSON, type: String, typ: Typ) = Builder(id, payload, type, typ)
+    }
+
+    class Builder(val id: String, val payload: JSON, val type: String, val typ: Typ) {
```

Comment:
I believe `typ` is always the same and equals to `application/didcomm-plain+json`, isn't it? So I would either remove it at all and use hard-coded value, or make it an optional param with `application/didcomm-plain+json` as a default.

Reply:
agree